cli/src/commands/file/annotate.rs

```diff
@@ -94,17 +101,27 @@ pub(crate) fn cmd_file_annotate(
 fn render_file_annotation(
     repo: &dyn Repo,
     ui: &mut Ui,
-    template_render: &TemplateRenderer<Commit>,
+    template_render: &TemplateRenderer<AnnotateData>,
     annotation: &FileAnnotation,
 ) -> Result<(), CommandError> {
     ui.request_pager();
     let mut formatter = ui.stdout_formatter();
+    let mut last_id = None;
     for (line_no, (commit_id, line)) in annotation.lines().enumerate() {
         let commit_id = commit_id.expect("should reached to the empty ancestor");
         let commit = repo.store().get_commit(commit_id)?;
-        template_render.format(&commit, formatter.as_mut())?;
-        write!(formatter, " {:>4}: ", line_no + 1)?;
+        let is_first_line = match last_id {
+            None => true,
+            Some(last) => last != commit_id,
+        };
+        let annotate_data = AnnotateData {
+            commit,
+            line_no: line_no + 1,
+            is_first_line,
+        };
+        template_render.format(&annotate_data, formatter.as_mut())?;
         formatter.write_all(line)?;
```

Comment:
Perhaps, we should also move `line` to template, but we don't have a `BString` template type right now. This can be addressed later.